### DISCUSSION TITLE
does registry have a "hasComponent" function?

### DISCUSSION THREAD

**Discussion by MrCheeseWasTaken** [skypjack/entt#861] [Q&A] [answered]
I was watching a tutorial on entt by The Cherno ([here](https://www.youtube.com/watch?v=D4hz0wEB978&t=1575s&ab_channel=TheCherno)) and at 23:59 he used `registry.has` which i tried but does seem to exist as my IDE says has is not a member of registry
Does registry have a "has" function the returns a bool?

**Comment by skypjack** [comment, marked as answer]
There are a couple of functions actually: `all_of` and `any_of`.

**Reply by MrCheeseWasTaken** [reply]
out of curiosity did registry use to have a 'has' but it got removed?

**Reply by skypjack** [reply]
Yeah, definitely. It got renamed. It had `has` and `any` but the latter was annoyingly conflicting with `entt::any`, so I renamed them. 😅
